src/components/FormFields.jsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+
+const typeValueMapping = {
+  file: e => e.target.files[0],
+  checkbox: e => e.target.checked,
+  text: e => e.target.value
+};
+
+export const Input = ({
+  id,
+  value,
+  type,
+  name,
+  groupId,
+  onChange = () => {},
+  error,
+  placeholder = "",
+  label,
+  initialChecked,
+  checked,
+  classes: { wrapperClass, inputClass, errorClass },
+  ...rest
+}) => {
+  const isTypePresentInMapping = !!typeValueMapping[type];
+  const newType = isTypePresentInMapping ? type : "text";
+  return (
+    <div className={wrapperClass}>
+      {label && <label htmlFor={id}>{label}</label>}
+      <input
+        {...rest}
+        id={id}
+        className={inputClass}
+        type={type}
+        value={value}
+        placeholder={placeholder}
+        name={name}
+        checked={type == "radio" || type == "checkbox" ? checked : undefined}
+        onChange={e => onChange(e, groupId || id, typeValueMapping[newType](e))}
```

Comment:
Do we need arrow function ?  Lets create a memoised arrow function. Also ensure that the input is changing. 
Lets just keep two params. On param to be an object (with custom data), other to be event

Reply:
For OnChange this should be args order: - key/id, value, event

Reply:
done